docs/design.md

```diff
@@ -1,92 +1,122 @@
-It is accessible via a thin [facade class](http://en.wikipedia.org/wiki/Facade_pattern) called [Logic](../src/main/java/teammates/logic/api/Logic.java) which makes use of several `*Logic` classes to handle the logic related to various types of data and to access data from the `Storage` component. In particular, `Logic` is responsible for these: 
-+ Managing relationships between entities. e.g., cascade logic for create/update/delete.
-+ Managing transactions. e.g., to ensure atomicity of a transaction.
-+ Sanitizing input values recevied from the UI component.
-+ Providing a mechanism for checking access control rights.
+1. Request forwarded to the `ControllerServlet`.
+1. `ControllerServlet` uses the `ActionFactory` to generate the matching `Action` object, e.g. `InstructorHomePageAction`.
+1. `ControllerServlet` executes the action.
+1. The `Action` object checks the access rights of the user. If the action is allowed, it will be performed, interacting with the `Logic` component as necessary.
+1. The `Action` gathers the data required for the page into a `PageData` object, e.g. `InstructorHomePageData`, creates a `ShowPageResult` object by enclosing the `PageData` object created previously, and returns it to the `ControllerServlet`.
+1. `ControllerServlet` sends the result. In case of a `ShowPageResult`, this is equivalent to forwarding to the matching JSP page.
+1. The JSP page uses the data in the given `PageData` object to generate the HTML page.
+1. The response will then be sent back to the browser, which will render the page.
+
+The other 30% types of request may differ in the following manners:
+- At times, the user will request for data asynchronously via AJAX. In such cases, the actions return an `AjaxResult`. `PageData` is also created for this request and the data obtained is pre-processed for display in the browser; that is, the result of the AJAX request is not the raw data.
+- After performing certain actions, the browser may load another page (redirects to another page), possibly with the status of the previous action. In such cases, the action returns a `RedirectResult`, which simply instructs the browser to send a fresh request for the specified page. In such cases, no `PageData` object will be generated for the original request.
+- The result of some actions is downloading of a file (e.g. a feedback session report) or an image (e.g. profile picture). In such cases, the result type will be `FileDownloadResult` or `ImageResult`, and no `PageData` object will be generated.
+- In some cases, the request is not forwarded to `ControllerServlet`, but to some other `*Servlet` as specified in `web.xml`. In such cases, the request will be processed as specified in those `*Servlet` objects; no common routine is observed in these other `*Servlet` objects.
+- Some requests may not forwarded to any `*Servlet` at all, but rather to a specified resource file of the web application (e.g. JSP files).
```

Comment:
Would it be helpful to add the below info?

Also, some requests for static files (e.g. css, js files) are served directly without going through `web.xml` at all. As this [GAE doc](https://cloud.google.com/appengine/docs/flexible/java/configuring-the-web-xml-deployment-descriptor) puts it,

> Note: Static files, files that are served verbatim to users such as images, CSS or JavaScript, are handled separately from paths mentioned in the deployment descriptor. A request for a URL path that matches a path to a file in the WAR that's considered a static file will serve the file, regardless of servlet and filter mappings in the deployment descriptor. You can exclude files from those treated as static files using the appengine-web.xml file.